### PR TITLE
Add support for verify peer.

### PR DESCRIPTION
**Pull request by conniey**
* Add logic to ConnectionHandler and support for verification options.
* Ensure that ClientOptions is not null.
* Removing MockReactorProviderHandler because you can mock concrete classes.

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java [hunk 1]
```diff
@@ -630,8 +645,10 @@ private ConnectionOptions getConnectionOptions() {
             ? CbsAuthorizationType.SHARED_ACCESS_SIGNATURE
             : CbsAuthorizationType.JSON_WEB_TOKEN;
 
+        final ClientOptions options = clientOptions != null ? clientOptions : new ClientOptions();
+
         return new ConnectionOptions(fullyQualifiedNamespace, credentials, authorizationType, transport, retryOptions,
-            proxyOptions, scheduler, null);
+            proxyOptions, scheduler, options, SslDomain.VerifyMode.VERIFY_PEER_NAME);
```
Comment (srnagar) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java:651]: Do we plan of supporting any other mode apart from VERIFY_PEER_NAME?

**Review by srnagar — APPROVED** [review]

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java (hunk 1 above)
Comment (conniey) [reply sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java:651]: In Service Bus, they had additional configuration via environment variables. It's not a feature we ever discussed.

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ConnectionOptions.java [hunk 2]
```diff
@@ -26,10 +27,13 @@
     private final String fullyQualifiedNamespace;
     private final CbsAuthorizationType authorizationType;
     private final ClientOptions clientOptions;
+    private final SslDomain.VerifyMode verifyMode;
 
     public ConnectionOptions(String fullyQualifiedNamespace, TokenCredential tokenCredential,
             CbsAuthorizationType authorizationType, AmqpTransportType transport, AmqpRetryOptions retryOptions,
-            ProxyOptions proxyOptions, Scheduler scheduler, ClientOptions clientOptions) {
+            ProxyOptions proxyOptions, Scheduler scheduler, ClientOptions clientOptions,
+        SslDomain.VerifyMode verifyMode) {
```
[the remark this answers is not in the document]
Comment (conniey) [reply sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ConnectionOptions.java:35]: oof. Good catch.

**Pull request merged**